string[]  values
float32[] probabilities
